libs/constrained-generators/src/Constrained/SumList.hs

```diff
@@ -0,0 +1,308 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Constrained.SumList where
+
+import Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NE
+import System.Random (Random (..))
+import Test.QuickCheck (Gen, choose, shuffle, vectorOf)
+
+-- =======================================================
+-- Helper functions for genSizedList
+
+data Solution t = Yes (NonEmpty [t]) | No [String]
+  deriving (Eq)
+
+instance Show t => Show (Solution t) where
+  show (No xs) = "No" ++ "\n" ++ unlines xs
+  show (Yes xs) = "Yes " ++ show xs
+
+-- | Similar to (concat :: [[x]] -> [x]) for lists
+concatNE :: NonEmpty (NonEmpty t) -> NonEmpty t
+concatNE (x :| []) = x
+concatNE (x :| (y : ys)) = x <> concatNE (y :| ys)
```

Comment:
You can just use [`sconcat`](https://downloads.haskell.org/ghc/9.10.1/docs/libraries/base-4.20.0.0-1f57/Data-Semigroup.html#v:sconcat)
```suggestion
```

Reply:
done.  Who knows all these esoteric functions? Alexey does! what luck.